constraint/pkg/client/client.go

```diff
@@ -214,19 +220,43 @@ func createTemplatePath(target, name string) string {
 
 // CreateCRD creates a CRD from template
 func (c *client) CreateCRD(ctx context.Context, templ *v1alpha1.ConstraintTemplate) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
-	if err := validateTargets(templ); err != nil {
+	nm := templ.ObjectMeta.Name
+	specCRD := &templ.Spec.CRD
+	targets := templ.Spec.Targets
+	if err := c.validateGenericCRD(nm, specCRD, targets); err != nil {
+		return nil, err
+	}
+	return c.createGenericCRD(nm, specCRD, targets, constraintGroup)
+}
+
+func (c *client) CreateMutationCRD(ctx context.Context, templ *v1alpha1.MutationTemplate) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
+	nm := templ.ObjectMeta.Name
+	specCRD := &templ.Spec.CRD
+	targets := templ.Spec.Targets
+	if err := c.validateGenericCRD(nm, specCRD, targets); err != nil {
 		return nil, err
 	}
-	if templ.ObjectMeta.Name == "" {
-		return nil, errors.New("Template has no name")
+	return c.createGenericCRD(nm, specCRD, targets, mutationGroup)
+}
+
+func (c *client) validateGenericCRD(objMetaName string, specCRD *v1alpha1.CRD, targets []v1alpha1.Target) error {
```

Comment:
specCRD should just be crd

Reply:
We had trouble here because `crd` is also the name of the thingy created by CreateCRD internally. So we needed to differentiate between the CRD from the spec we were passing in, and the one that got passed back. `specCRD` seemed like a decent compromise and we used it in multiple places.